### DISCUSSION TITLE
Calling a custom function from a button press in a mux

### DISCUSSION THREAD

**Discussion by MoSaye** [tttapa/Control-Surface#966] [Q&A] [answered]
Hello there, 

Is there a way to call a function when a single button is pressed, the button is connected in a mux, so what i want to do is to call a custom function when this specific button is pressed and sends a midi note on

Thanks for your help and great work

**Comment by MoSaye** [comment]
so for example this is the mux code:

NoteButton LimitSwitches[] = {
  {mux3.pin(0), {note:0x00, CHANNEL_1}},
  {mux3.pin(1), {note:0x01, CHANNEL_1}},
  {mux3.pin(2), {note:0x02, CHANNEL_1}},
  {mux3.pin(3), {note:0x03, CHANNEL_1}},
  {mux3.pin(4), {note:0x04, CHANNEL_1}},
..........

I want to call a function called x for example when the first button is pressed, or the third button, just a specific button in the mux array

**Comment by tttapa** [comment, marked as answer]
You can get the button state from the `NoteButton`s:

```cpp
if (LimitSwitches[i].getButtonState() == Button::Falling)
  someFunction();
```
If you need more flexibility, see this example: https://tttapa.github.io/Control-Surface-doc/Doxygen/d4/d3b/Custom-MIDI-Output-Element_8ino-example.html

**Reply by MoSaye** [reply]
Thank you for the explanation, getting the button state works fine and i will look into the custom midi output for more.
